clubs_guides/views.py

```diff
@@ -1,3 +1,73 @@
-from django.shortcuts import render
+from rest_framework.generics import ListAPIView, RetrieveAPIView
 
-# Create your views here.
+from clubs_guides.models import ClubsGuide, Category
+from clubs_guides.serializers import (
+    ClubsGuideSerializer,
+    CategorySerializer,
+    CategoryWithClubsGuideSerializer,
+)
+
+
+class ClubsGuidesListView(ListAPIView):
+    """
+    A view that permits a GET to allow listing all the clubs guides
+    in the database
+
+    **Route** - `/clubsguides`
+
+    """
+    queryset = ClubsGuide.objects.all()
+    serializer_class = ClubsGuideSerializer
+
+
+class ClubsGuideView(RetrieveAPIView):
+    """
+    A view that permits a GET to allow listing of a single clubs guide
+    by providing its id.
+
+    **Route** - `/clubsguides/:id`
```

Comment:
Hmm I'm apprehensive about this approach because it seems a bit anti-pattern. A view can really apply to multiple url patterns so this would break that case which is why a view is never aware of its url bindings.

I think I will hold off on this until we discuss this later to see what we can do about documenting better for frameworks in general